tests/linearizability/model.go

```diff
@@ -104,50 +104,65 @@ func step(state EtcdState, request etcdRequest, response etcdResponse) (bool, Et
 	}
 }
 
-func initState(request etcdRequest, response etcdResponse) EtcdState {
+func initState(request EtcdRequest, response EtcdResponse) EtcdState {
 	state := EtcdState{
-		Key:          request.key,
-		LastRevision: response.revision,
-		FailedWrites: map[string]struct{}{},
+		Key:          request.Key,
+		LastRevision: response.Revision,
 	}
-	switch request.op {
+	switch request.Op {
 	case Get:
-		state.Value = response.getData
+		state.Value = response.GetData
 	case Put:
-		if response.err == nil {
-			state.Value = request.putData
+		if response.Err == nil {
+			state.Value = request.PutData
 		} else {
-			state.FailedWrites[request.putData] = struct{}{}
+			state.FailedWrite = &request
 		}
 	default:
 		panic("Unknown operation")
 	}
 	return state
 }
 
-func stepGet(state EtcdState, request etcdRequest, response etcdResponse) (bool, EtcdState) {
-	if state.Value == response.getData && state.LastRevision <= response.revision {
+func stepGet(state EtcdState, request EtcdRequest, response EtcdResponse) (bool, EtcdState) {
+	if state.Value == response.GetData && state.LastRevision == response.Revision {
+		state.LastRevision = response.Revision
+		state.FailedWrite = nil
 		return true, state
 	}
-	_, ok := state.FailedWrites[response.getData]
-	if ok && state.LastRevision < response.revision {
-		state.Value = response.getData
-		state.LastRevision = response.revision
-		delete(state.FailedWrites, response.getData)
-		return true, state
+	if state.FailedWrite != nil && state.LastRevision < response.Revision {
```

Comment:
```suggestion
	if state.FailedWrite != nil && state.LastRevision <= response.Revision {
```

Reply:
> writes always increase revision.

With the addition of delete operation, it doesn't stand anymore. Deleting an non-exist will not increase the revision.

Reply:
https://github.com/etcd-io/etcd/pull/14802/files#diff-f6cd9cfbf4b5310cea9742f346927212c88f39b5cab928a434fe3ffb44d50cccR182-R187

Reply:
Yes, but that should be changed by PR that introduced delete.